### PR TITLE
don't install configmaps on no grafana configuration

### PR DESCRIPTION
**Pull request by AjayTripathy**

### CODE REVIEW

File: cost-analyzer/templates/grafana-dashboard-cluster-metrics-template.yaml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+{{- if .Values.grafana -}}
+{{- if .Values.grafana.sidecar -}}
+{{- if .Values.grafana.dashboards -}}
+{{- if .Values.grafana.dashboards.enabled -}}
```
Comment (Nuru) [inline cost-analyzer/templates/grafana-dashboard-cluster-metrics-template.yaml]: This is undesirable overkill
```
{{- if .Values.grafana -}}
{{- if .Values.grafana.sidecar -}}
{{- if .Values.grafana.dashboards -}}
{{- if .Values.grafana.dashboards.enabled -}}
```
All we need or want for the dashboards is `{{- if .Values.grafana.dashboards.enabled -}}`

If you want to get fancy, you can also add a value specific to each dashboard, such as `{{- if .Values.grafana.dashboards.clusterMetrics.enabled -}}`

File: cost-analyzer/templates/grafana-datasource-template.yaml [hunk 2]
```diff
@@ -1,3 +1,7 @@
+{{- if .Values.grafana -}}
+{{- if .Values.grafana.sidecar -}}
+{{- if .Values.grafana.datasources -}}
+{{- if .Values.grafana.datasources.enabled -}}
 apiVersion: v1
```
Comment (Nuru) [inline cost-analyzer/templates/grafana-datasource-template.yaml]: Like with the dashboards, all we need for the datasource is `{{- if .Values.grafana.datasources.enabled -}}`; the other conditionals are not helpful.

**Review by Nuru — CHANGES_REQUESTED** [review]

File: cost-analyzer/templates/grafana-dashboard-cluster-metrics-template.yaml (hunk 1 above)
Comment (AjayTripathy) [reply cost-analyzer/templates/grafana-dashboard-cluster-metrics-template.yaml]: These conditionals are defensive programming to protect people using --reuse-values from earlier values.yaml, which we in general support in version 1.X.X. Supplying a default isn't enough We can look into simplifying in a later version.

File: cost-analyzer/templates/grafana-datasource-template.yaml (hunk 2 above)
Comment (AjayTripathy) [reply cost-analyzer/templates/grafana-datasource-template.yaml]: Ditto re: unnecessary conditionals. However, this does raise a point about proxying to grafana and assumptions we've made about grafana being reachable-- I've addressed those in another commit.

**Review by Nuru — APPROVED** [review]

File: cost-analyzer/templates/grafana-dashboard-cluster-metrics-template.yaml (hunk 1 above)
Comment (Nuru) [reply cost-analyzer/templates/grafana-dashboard-cluster-metrics-template.yaml]: @AjayTripathy okay

**Pull request merged**